src/v/cloud_storage/topic_mount_handler.cc

```diff
@@ -155,6 +157,45 @@ ss::future<topic_unmount_result> topic_mount_handler::unmount_topic(
     co_return topic_unmount_result::success;
 }
 
+ss::future<result<chunked_vector<labeled_namespaced_topic>>>
+topic_mount_handler::list_unmounted_topics(retry_chain_node& parent) {
+    auto log = retry_chain_logger(cst_log, parent);
+
+    cloud_storage_clients::object_key prefix{
+      cloud_storage::topic_mount_manifest_path::prefix()};
+
+    vlog(log.debug, "listing unmounted topics with prefix {}", prefix);
+
+    auto list_result = co_await _remote.list_objects(_bucket, parent, prefix);
+    if (!list_result) {
+        vlog(log.error, "failed to list objects: {}", list_result.error());
+        co_return list_result.error();
+    }
+
+    chunked_vector<labeled_namespaced_topic> ret;
+    for (const auto& item : list_result.assume_value().contents) {
+        vlog(log.trace, "parsing object {}", item.key);
+
+        auto path_parse_result
+          = cloud_storage::topic_mount_manifest_path::parse(
+            std::string(item.key));
+        if (!path_parse_result) {
+            vlog(log.error, "failed to parse object key {}", item.key);
+            continue;
+        }
+
+        ret.emplace_back(labeled_namespaced_topic{
+          .label = cloud_storage::remote_label(
+            path_parse_result->cluster_uuid()),
```

Comment:
@nvartolomei Willem is adding revision to the _mount_ manifest path in https://github.com/redpanda-data/redpanda/pull/23730. Once it is merged, can we return `{uuid}/{rev}` here? The plan is to support a hint from user when mounting, and the hint is a prefix for this path component of the _topic_ manifest. If there are multiple unmounted topics with the same name and cluster UUID having a revision would be beneficial.

Reply:
Should we keep them separate and ask user 3 fields (label (cluster uuid), topic, revision)? Because that's how things are represented in most parts. I'm afraid that putting together the label/cluster uuid with revision id will cause confusion long term... It is an accident for the most part that label and rev id are next to each other here (topic manifest). I.e. partition manifest, the path is `<label>/meta/<ns>/<topic>/<partition>_<revision>` and who knows what we'll come up with down the road.